Move remaining ivars from WKWebViewConfiguration to API::PageConfiguration

achristensen07 opened the pull request:
#### cf4a98a316fb56c1d99211c7d5542233c8784398
<pre>
Move remaining ivars from WKWebViewConfiguration to API::PageConfiguration
<a href="https://bugs.webkit.org/show_bug.cgi?id=271451">https://bugs.webkit.org/show_bug.cgi?id=271451</a>
<a href="https://rdar.apple.com/125216495">rdar://125216495</a>

Reviewed by Tim Horton.

* Source/WebKit/Shared/API/Cocoa/WKDataDetectorTypesInternal.h:
(toWKDataDetectorTypes):
* Source/WebKit/UIProcess/API/APIPageConfiguration.cpp:
(API::PageConfiguration::setRelatedPage):
(API::PageConfiguration::setPageToCloneSessionStorageFrom):
(API::PageConfiguration::alternateWebViewForNavigationGestures const):
(API::PageConfiguration::setAlternateWebViewForNavigationGestures):
* Source/WebKit/UIProcess/API/APIPageConfiguration.h:
(API::PageConfiguration::dataDetectorTypes const):
(API::PageConfiguration::setDataDetectorTypes):
(API::PageConfiguration::selectionGranularity const):
(API::PageConfiguration::setSelectionGranularity):
(API::PageConfiguration::allowsPictureInPictureMediaPlayback const):
(API::PageConfiguration::setAllowsPictureInPictureMediaPlayback):
(API::PageConfiguration::mediaTypesRequiringUserActionForPlayback const):
(API::PageConfiguration::setMediaTypesRequiringUserActionForPlayback):
(API::PageConfiguration::suppressesIncrementalRendering const):
(API::PageConfiguration::setSuppressesIncrementalRendering):
(API::PageConfiguration::allowsAirPlayForMediaPlayback const):
(API::PageConfiguration::setAllowsAirPlayForMediaPlayback):
(API::PageConfiguration::ignoresViewportScaleLimits const):
(API::PageConfiguration::setIgnoresViewportScaleLimits):
(API::PageConfiguration::userInterfaceDirectionPolicy const):
(API::PageConfiguration::setUserInterfaceDirectionPolicy):
* Source/WebKit/UIProcess/API/Cocoa/APIPageConfigurationCocoa.mm:
(API::PageConfiguration::Data::defaultMediaTypesRequiringUserActionForPlayback):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfiguration.mm:
(-[WKWebViewConfiguration init]):
(-[WKWebViewConfiguration selectionGranularity]):
(-[WKWebViewConfiguration setSelectionGranularity:]):
(-[WKWebViewConfiguration setAllowsPictureInPictureMediaPlayback:]):
(-[WKWebViewConfiguration allowsPictureInPictureMediaPlayback]):
(-[WKWebViewConfiguration setIgnoresViewportScaleLimits:]):
(-[WKWebViewConfiguration ignoresViewportScaleLimits]):
(-[WKWebViewConfiguration setDataDetectorTypes:]):
(-[WKWebViewConfiguration dataDetectorTypes]):
(-[WKWebViewConfiguration userInterfaceDirectionPolicy]):
(-[WKWebViewConfiguration setUserInterfaceDirectionPolicy:]):
(-[WKWebViewConfiguration initWithCoder:]):
(-[WKWebViewConfiguration copyWithZone:]):
(-[WKWebViewConfiguration mediaTypesRequiringUserActionForPlayback]):
(-[WKWebViewConfiguration setMediaTypesRequiringUserActionForPlayback:]):
(-[WKWebViewConfiguration suppressesIncrementalRendering]):
(-[WKWebViewConfiguration setSuppressesIncrementalRendering:]):
(-[WKWebViewConfiguration setAllowsAirPlayForMediaPlayback:]):
(-[WKWebViewConfiguration allowsAirPlayForMediaPlayback]):
(-[WKWebViewConfiguration _relatedWebView]):
(-[WKWebViewConfiguration _setRelatedWebView:]):
(-[WKWebViewConfiguration _webViewToCloneSessionStorageFrom]):
(-[WKWebViewConfiguration _setWebViewToCloneSessionStorageFrom:]):
(-[WKWebViewConfiguration _alternateWebViewForNavigationGestures]):
(-[WKWebViewConfiguration _setAlternateWebViewForNavigationGestures:]):

Canonical link: <a href="https://commits.webkit.org/276554@main">https://commits.webkit.org/276554@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/345c2c02046df9115d9f32fc5f4be7f9282151d2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/44957 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/24066 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/47456 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/47615 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/40964 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/28136 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/21464 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/36905 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/45535 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/21125 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/38715 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/17991 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/18549 "Passed tests") | | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/3005 "Built successfully") | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/40150 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/49287 "Built successfully") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/19929 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/16478 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/43903 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/21245 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/42672 "Passed tests") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/21586 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6253 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/20924 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WebKit/UIProcess/API/APIPageConfiguration.h:
```diff
@@ -407,8 +445,9 @@ class PageConfiguration : public ObjectImpl<Object::Type::PageConfiguration> {
         WeakPtr<WebKit::WebExtensionController> weakWebExtensionController;
 #endif
         RefPtr<WebKit::WebPageGroup> pageGroup;
-        RefPtr<WebKit::WebPageProxy> relatedPage;
```
Isn't this a potentially somewhat large and API-observable behavior change?

achristensen07 replied:
No, WKWebViewConfiguration already has it as a WeakObjCPtr, so making API::PageConfiguration's relatedPage a WeakPtr is preserving API behavior, though only indirectly observable.  This only changes the behavior of WKPageConfigurationGetRelatedPage to match WKWebViewConfiguration, and that SPI has no users.

achristensen07 committed:
Move remaining ivars from WKWebViewConfiguration to API::PageConfiguration
https://bugs.webkit.org/show_bug.cgi?id=271451
rdar://125216495

Reviewed by Tim Horton.

* Source/WebKit/Shared/API/Cocoa/WKDataDetectorTypesInternal.h:
(toWKDataDetectorTypes):
* Source/WebKit/UIProcess/API/APIPageConfiguration.cpp:
(API::PageConfiguration::setRelatedPage):
(API::PageConfiguration::setPageToCloneSessionStorageFrom):
(API::PageConfiguration::alternateWebViewForNavigationGestures const):
(API::PageConfiguration::setAlternateWebViewForNavigationGestures):
* Source/WebKit/UIProcess/API/APIPageConfiguration.h:
(API::PageConfiguration::dataDetectorTypes const):
(API::PageConfiguration::setDataDetectorTypes):
(API::PageConfiguration::selectionGranularity const):
(API::PageConfiguration::setSelectionGranularity):
(API::PageConfiguration::allowsPictureInPictureMediaPlayback const):
(API::PageConfiguration::setAllowsPictureInPictureMediaPlayback):
(API::PageConfiguration::mediaTypesRequiringUserActionForPlayback const):
(API::PageConfiguration::setMediaTypesRequiringUserActionForPlayback):
(API::PageConfiguration::suppressesIncrementalRendering const):
(API::PageConfiguration::setSuppressesIncrementalRendering):
(API::PageConfiguration::allowsAirPlayForMediaPlayback const):
(API::PageConfiguration::setAllowsAirPlayForMediaPlayback):
(API::PageConfiguration::ignoresViewportScaleLimits const):
(API::PageConfiguration::setIgnoresViewportScaleLimits):
(API::PageConfiguration::userInterfaceDirectionPolicy const):
(API::PageConfiguration::setUserInterfaceDirectionPolicy):
* Source/WebKit/UIProcess/API/Cocoa/APIPageConfigurationCocoa.mm:
(API::PageConfiguration::Data::defaultMediaTypesRequiringUserActionForPlayback):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfiguration.mm:
(-[WKWebViewConfiguration init]):
(-[WKWebViewConfiguration selectionGranularity]):
(-[WKWebViewConfiguration setSelectionGranularity:]):
(-[WKWebViewConfiguration setAllowsPictureInPictureMediaPlayback:]):
(-[WKWebViewConfiguration allowsPictureInPictureMediaPlayback]):
(-[WKWebViewConfiguration setIgnoresViewportScaleLimits:]):
(-[WKWebViewConfiguration ignoresViewportScaleLimits]):
(-[WKWebViewConfiguration setDataDetectorTypes:]):
(-[WKWebViewConfiguration dataDetectorTypes]):
(-[WKWebViewConfiguration userInterfaceDirectionPolicy]):
(-[WKWebViewConfiguration setUserInterfaceDirectionPolicy:]):
(-[WKWebViewConfiguration initWithCoder:]):
(-[WKWebViewConfiguration copyWithZone:]):
(-[WKWebViewConfiguration mediaTypesRequiringUserActionForPlayback]):
(-[WKWebViewConfiguration setMediaTypesRequiringUserActionForPlayback:]):
(-[WKWebViewConfiguration suppressesIncrementalRendering]):
(-[WKWebViewConfiguration setSuppressesIncrementalRendering:]):
(-[WKWebViewConfiguration setAllowsAirPlayForMediaPlayback:]):
(-[WKWebViewConfiguration allowsAirPlayForMediaPlayback]):
(-[WKWebViewConfiguration _relatedWebView]):
(-[WKWebViewConfiguration _setRelatedWebView:]):
(-[WKWebViewConfiguration _webViewToCloneSessionStorageFrom]):
(-[WKWebViewConfiguration _setWebViewToCloneSessionStorageFrom:]):
(-[WKWebViewConfiguration _alternateWebViewForNavigationGestures]):
(-[WKWebViewConfiguration _setAlternateWebViewForNavigationGestures:]):

Canonical link: https://commits.webkit.org/276554@main
